Gemfile

```diff
@@ -45,7 +47,7 @@ gem "more_core_extensions",           "~>3.3"
 gem "nakayoshi_fork",                 "~>0.0.3"  # provides a more CoW friendly fork (GC a few times before fork)
 gem "net-ldap",                       "~>0.14.0",      :require => false
 gem "net-ping",                       "~>1.7.4",       :require => false
-gem "net-ssh",                        "=3.2.0",        :require => false
+gem "net-ssh",                        "=4.2.0",        :require => false
```

Comment:
Can we make this more permissive?

It looks like we planned on making it less restrictive in https://github.com/ManageIQ/manageiq/pull/9284

Opinions on what this should look like @jrafanie @Fryguy @bdunne ?

Reply:
@carbonin Sure, makes sense to allow updates. I will wait for others' opinions.

Reply:
```
$ git grep -i "net::ssh"
app/models/host.rb:784:    rescue Net::SSH::AuthenticationFailed
app/models/host.rb:786:    rescue Net::SSH::HostKeyMismatch
app/models/host.rb:1420:        rescue Net::SSH::HostKeyMismatch
```

I'd change the version requirement here to `">=3.2.0", "<5"` then we can maintain the dependency on `linux_admin`.

Reply:
Technically `manageiq-gems-pending` should claim it as a dependency too until everything is moved to `linux_admin`

Reply:
It's not clear from this PR which version fixes this problem for ruby 2.4.  The linked net-ssh issue seems to indicate 4.1 fixes it.  

Note, there was some old ssh code to access hypervisor patch and OS information, I'm not sure who owns that now @gmcculloug.  I don't know that it's easy to verify.  I'm not familiar how to test the container deployment code either, cc @alongoldboim.  I'm not sure if we have other ssh consumers.

I'm fine with upgrading and doing `~>4.2.0` but it would be good to list what we're getting when we upgrade and maybe get a 👍  from people familiar with the other ssh use cases.

Reply:
@jrafanie yeah, I did not include info what version solved that issue. I tested it (credential user+ssh key validation and smart state analysis with OpenStack Infra provider) and in fact it works (for me) with 4.0.0, 4.1.0 and 4.2.0.

I did not find in net-ssh's changelog what solved the issue though.

And I for sure understand that we need approval from others who are using it so we are sure that it works correctly and nothing is broken by bumping major version of net-ssh.